client/components/token-field/suggestions-list.jsx

```diff
@@ -86,13 +91,42 @@ class SuggestionsList extends React.PureComponent {
 		);
 	}
 
+	_removeEmptyLabelsFromSuggestions( suggestions ) {
+		const filteredSuggestions = [];
+		for ( let i = 0; i < suggestions.length; i++ ) {
+			if (
+				isSuggestionLabel( suggestions[ i ] ) &&
+				( ( suggestions?.[ i + 1 ] && isSuggestionLabel( suggestions[ i + 1 ] ) ) ||
+					! suggestions?.[ i + 1 ] )
+			) {
+				continue;
+			}
+			filteredSuggestions.push( suggestions[ i ] );
+		}
+		return filteredSuggestions;
+	}
+
 	_renderSuggestions = () => {
-		return map( this.props.suggestions, ( suggestion, index ) => {
-			const match = this._computeSuggestionMatch( suggestion );
+		const filteredSuggestions = this._removeEmptyLabelsFromSuggestions( this.props.suggestions );
+
+		return map( filteredSuggestions, ( suggestion, index ) => {
```

Comment:
You could use array's `.map` instead of lodash `map`:
```suggestion
		return filteredSuggestions.map( ( suggestion, index ) => {
```